addons/Popochiu/Engine/Objects/Character/PopochiuCharacter.gd

```diff
@@ -240,19 +256,75 @@ func _get_vo_cue(emotion := '') -> String:
 			return cue_name
 	return ''
 
+func face_direction(destination: Vector2, is_in_queue := true):
+	# Get the vector from the origin to the destination.
+	var vectX = destination.x - position.x
+	var vectY = destination.y - position.y
+	# Determine the angle of the movement vector.
+	var rad = atan2(vectY, vectX)
+	var angle = rad2deg(rad)
+	# Tolerance in degrees, to avoid U D L R are only
+	# achieved on precise angles such as 0 90 180 deg.
+	var t = 20
+	# Determine the direction the character is facing.
+	# Remember: Y coordinates have opposite sign in Godot.
+	# this means that negative angles are up movements.
+	# Set the direction using the _looking property.
+	# We cannot use the face_* functions because they
+	# set the state as IDLE.
+	if angle >= -(0 + t) and angle < (0 + t):
+		_looking_dir = LOOKING.RIGHT
+	elif angle > (0 + t) and angle < (90 - t):
+		_looking_dir = LOOKING.DOWN_RIGHT
+	elif angle > (90 - t) and angle < (90 + t):
+		_looking_dir = LOOKING.DOWN
+	elif angle > (90 + t) and angle < (180 - t):
+		_looking_dir = LOOKING.DOWN_LEFT
+	elif angle >= (180 - t) or angle <= -(180 -t ):
+		_looking_dir = LOOKING.LEFT
+	elif angle <= -(0 + t) and angle > -(90 - t):
+		_looking_dir = LOOKING.UP_RIGHT
+	elif angle < -(90 - t) and angle > -(90 + t):
+		_looking_dir = LOOKING.UP
+	elif angle < -(90 + t) and angle > -(180 - t):
+		_looking_dir = LOOKING.UP_LEFT
+
+func play_animation(animation_label: String, animation_fallback := 'idle', blocking := false):
+	if has_node("AnimationPlayer"):
+		# Check if animation is null in case it is call dynamically.
+		if animation_label == null:
+			$AnimationPlayer.stop()
+			return
+		if $AnimationPlayer.has_animation(animation_label):
+			$AnimationPlayer.play(animation_label)
+		else:
+		   $AnimationPlayer.play(animation_fallback)
 
-# ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ VIRTUAL ░░░░
 func play_idle() -> void:
-	pass
-
+	play_animation('idle');
 
 func play_walk(target_pos: Vector2) -> void:
-	pass
-
+	# Set the default parameters for play_animation()
+	var animation_label = 'walk'
+	var animation_fallback = 'idle'
+	# If facing any direction that is not up or down
+	# use an horizontal walking animation, or suggest
+	# the use of a proper animation for up and down.
+	if _looking_dir in [LOOKING.LEFT, LOOKING.RIGHT, LOOKING.UP_LEFT, LOOKING.UP_RIGHT, LOOKING.DOWN_LEFT, LOOKING.DOWN_RIGHT]:
+		animation_label = 'walk'
+	else:
+		animation_fallback = 'walk'
+		match _looking_dir:
+			LOOKING.DOWN: animation_label = 'walk_f'
+			LOOKING.UP: animation_label = 'walk_b'
```

Comment:
I would keep the animation suffix consistend with "down" and "up", so `walk_d' and `walk_u`.

Reply:
Good point! I'm fixing this